pkg/cloudprovider/aws/aws.go

```diff
@@ -73,12 +107,65 @@ func (f *ec2InstanceFilter) Matches(instance ec2.Instance) bool {
 }
 
 // goamzEC2 is an implementation of the EC2 interface, backed by goamz
-type GoamzEC2 struct {
-	ec2 *ec2.EC2
+type goamzEC2 struct {
+	auth aws.Auth
+	ec2  *ec2.EC2
+
+	mutex      sync.Mutex
+	elbClients map[string]*elb.ELB
+}
+
+func newGoamzEC2(auth aws.Auth, regionName string) (*goamzEC2, error) {
+	region, ok := aws.Regions[regionName]
+	if !ok {
+		return nil, fmt.Errorf("not a valid AWS region: %s", regionName)
+	}
+
+	self := &goamzEC2{}
+	self.ec2 = ec2.New(auth, region)
+	self.auth = auth
+	self.elbClients = make(map[string]*elb.ELB)
+	return self, nil
+}
+
+// Find the kubernetes vpc
+func (self *goamzEC2) ListVpcs(filterName string) ([]ec2.VPC, error) {
+	client := self.ec2
+
+	// TODO: How do we want to identify our VPC?
+	filter := ec2.NewFilter()
+	filter.Add("tag:Name", filterName)
+
+	ids := []string{}
+	response, err := client.DescribeVpcs(ids, filter)
+	if err != nil {
+		glog.Error("error listing VPCs", err)
+		return nil, err
+	}
+
+	vpcs := response.VPCs
```

Comment:
You're right - it doesn't buy us anything right now - it's more of a statement of intent!

I'm working towards replacing goamz entirely (#6162).  I'm also heading towards the idea of returning a wrapper object here (awsVpc or awsKubernetesCluster) which would encapsulate some of the logic for dealing with AWS.